app/validation/validators.py

```diff
@@ -167,3 +150,47 @@ def __call__(self, form, from_field, to_field):
             raise validators.ValidationError(self.messages['INVALID_DATE_RANGE_TO_FROM_SAME'])
         elif from_date > to_date:
             raise validators.ValidationError(self.messages['INVALID_DATE_RANGE_TO_BEFORE_FROM'])
+
+
+class FormResponseRequired(object):
+    field_flags = ('required', )
+
+    def __init__(self, message=None):
+        if not message:
+            message = error_messages['MANDATORY']
+        self.message = message
+
+    def __call__(self, form, field):
+
+        if hasattr(form, 'day'):
```

Comment:
or use the any() built-in to do the what Ian suggests, but will exit as soon as any field value evaluates to False without evaluating the whole list (I don't think the cast to bool is needed this way as well):

```python
if not any(field.data for field in form):
    raise validators.StopValidation(self.message)
```